reactor-netty-core/src/main/java/reactor/netty/transport/TransportConnector.java

```diff
@@ -219,7 +220,31 @@ static void doConnect(SocketAddress remoteAddress, @Nullable Supplier<? extends
 				return monoChannelPromise;
 			}
 
+			if (config instanceof ClientTransportConfig) {
+				final ClientTransportConfig clientTransportConfig = (ClientTransportConfig)config;
+				if (clientTransportConfig.doOnResolve != null) {
+					clientTransportConfig.doOnResolve.accept(Connection.from(channel));
+				}
+			}
+
 			Future<SocketAddress> resolveFuture = resolver.resolve(remoteAddress);
+
+			if (config instanceof ClientTransportConfig) {
+				final ClientTransportConfig clientTransportConfig = (ClientTransportConfig)config;
+				resolveFuture.addListener((FutureListener<SocketAddress>) future -> {
+					if (future.cause() != null) {
+						if (clientTransportConfig.doOnResolveError != null) {
+							clientTransportConfig.doOnResolveError.accept(Connection.from(channel), future.cause());
+						}
+					}
+					else {
+						if (clientTransportConfig.doAfterResolve != null) {
+							clientTransportConfig.doAfterResolve.accept(Connection.from(channel), future.getNow());
+						}
+					}
+				});
+			}
+
```

Comment:
I was thinking that here we can use `config.connectionObserver()` and send events instead of casting to a specific config but with the current `ConnectionObserver` we cannot propagate `Connection` + `SocketAddress`. (this was why I mentioned `ConnectionObserver` in my previous comment)